### PR TITLE
Feature/reciperepo class

### LINKED ISSUE #14: Build 'RecipeRepository' class

### PR DESCRIPTION
**Pull request by philalewis**
What (if any) features are you implementing?
- Write out RecipeRepo class
- Refactor test files to match expectations
- Add filter methods to the class

What (if anything) did you refactor?
- Multiple test files

Were there any issues that arose?
- We need to be able to account for upper and lower case letters in searches

Is there anything that you need from your teammate?
- Review code and add comments where necessary. Merge if all looks good

Any other comments, questions, or concerns?
- No

### CODE REVIEW

File: src/classes/RecipeRepository.js [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
+    this.currentRecipes = this.currentRecipes.filter(recipe => recipe.tags.includes(tag));
+  }
+
+  filterRecipesByName() {
+    this.currentRecipes = this.currentRecipes.filter(recipe => {
+      return recipe.name.includes(this.filterTerm);
+    })
+  }
+
+  filterRecipesByIngredient() {
```
Comment (mr-ryan12) [inline src/classes/RecipeRepository.js:27]: We'll have to account for upper/lower case at user input.

File: test/RecipeRepository-test.js [hunk 2]
```diff
@@ -28,65 +31,71 @@ describe('RecipeRepo', () => {
...
+    expect(recipeRepo.filterTerm).to.equal('dinner');
   });
 
-  it('should be able to return a filtered list of recipe ids', () => {
-    expect(recipeRepo.currentRecipeIds.length).to.be(2);
-    recipeRepo.addFilter('snack');
-    recipeRepo.filterRecipes();
-    expect(recipeRepo.currentRecipeIds).to.be.an('array');
-    expect(recipeRepo.currentRecipeIds[0]).to.equal(10);
+  it('should be able to filter recipes by ingredient', () => {
```
Comment (mr-ryan12) [inline test/RecipeRepository-test.js:65]: Nicely done getting all the tests to pass.

File: test/RecipeRepository-test.js [hunk 3]
```diff
@@ -28,65 +31,71 @@ describe('RecipeRepo', () => {
...
+    recipeRepo.filterRecipesByTag('sandwich');
+    expect(recipeRepo.currentRecipes.length).to.equal(2);
     recipeRepo.clearFilters();
-    expect(recipeRepo.currentRecipeIds.length).to.equal(2);
+    recipeRepo.filterRecipesByTag('dinner');
+    expect(recipeRepo.currentRecipes.length).to.equal(1);
+  });
 
+  it('should be able to clear search terms', () => {
+    recipeRepo.addFilter('Burger');
```
Comment (oacortez) [inline test/RecipeRepository-test.js:90]: Will need to discuss and figure out to upper or lower case on this scenario.

File: test/RecipeRepository-test.js [hunk 4]
```diff
@@ -28,65 +31,71 @@ describe('RecipeRepo', () => {
...
   
   it('should be class instances', () => {
     expect(recipeRepo.recipes[0]).to.be.an.instanceof(Recipe);
-    expect(recipeRepo.recipes[1].ingredients).to.be.an.instanceof(Ingredient);
+    expect(recipeRepo.recipes[1].ingredients[0]).to.be.an.instanceof(Ingredient);
   })
 
   it('should have a list of current recipes', () => {
     expect(recipeRepo.cookbookRecipes).to.be.an('array');
-    expect(recipeRepo.cookbookRecipes.length).to.be(0);
```
Comment (mr-ryan12) [inline test/RecipeRepository-test.js:41]: Nice catch.

**Pull request merged**